### ISSUE TITLE
Feature Request - Add Dialog polyfill

### ISSUE THREAD

**Issue by JamesCoyle** [Financial-Times/polyfill-service#2057]
The dialog element is now defined in the [HTML 5.2 specification](https://www.w3.org/TR/html52/interactive-elements.html#the-dialog-element). It is a really useful element that everyone should be using for modal elements as it has many benefits for accessibility and prevents any annoying z-index issues.

Can a polyfill be added for browsers that do not yet implement it? I believe this is the defacto polyfill for the dialog element: https://github.com/GoogleChrome/dialog-polyfill

**Comment by Cedric-ruiu** [comment]
+1
